TechnicalDocuments/Proposals/BabylonDesignLibrary.md

```diff
@@ -0,0 +1,215 @@
+We can introduce `DesignLibrary` which will be responsible for creating these components following
+the semantics of the design styleguide which will be provided to us.
+
+The purpose of  `DesignLibrary` isn't to expose the full potential of `Bento` components but to 
+follow specific design guidelines.
+As an example of this is: 
+A `Bento` component supports different behaviours like  `Deletable` the goal of this
+proposal isn't to replicate them or to make a wrapper around them but to follow specific design guidelines
+and to offer reusable UI elements.
+
+Also with `DesignLibrary` we want to limit the exposure of specific components, in order to be able to introduce  
+drastic design changes in the future as easy as possible.
+For example `DesignLibrary.ComponentsBuilder.button` returns `AnyRenderable` instead of `Component.Button`
+because in the future we may decide that we need to use a different button component. We want to depend on abstractions.
+
+Furthermore, `DesignLibrary` can also host alerts, custom viewcontroller animations, etc.
+
+``` swift
+
+struct DesignLibrary {
+    let tokens: Tokens
+
+    var viewControllerBuilder: ViewControllerBuilder {
+        return ViewControllerBuilder(tokens: Tokens)
+    }
+
+    var componentsBuilder: ComponentsBuilder {
+        return ComponentsBuilder(tokens: Tokens, traits: UITraitCollection())
+    }
+
+    init(
+        tokens: Tokens
+    ) {
+        self.tokens = tokens
+    }
+}
+
+extension DesignLibrary {
+    struct Tokens {
+        let typography: Typography
+        let colors: Colors
+    }
+}
+
+extension DesignLibrary.Tokens {
+    struct Typography {
+        let headline1: UIFont
+        let headline2: UIFont
+        let body1: UIFont
+        let body2: UIFont
+    }
+
+    struct Typography {
+        struct FontAttributes {
+            let textStyle: UIFont.TextStyle
+            let weight: UIFont.Weight
+        }
+
+        let title1: FontAttributes
+        let title2: FontAttributes
+        let headline: FontAttributes
+    }
+
+    struct Colors {
+        let primary: UIColor
+        let secondary: UIColor
+    }
+}
+
+extension DesignLibrary {
+    struct ComponentsBuilder: BoxAppearance {
+        let tokens: Tokens
+        var traits: UITraitCollection
+
+        @available(*, deprecated, message: "Use DesignLibrary.Tokens")
+        public var brandColors: BrandColorsProtocol = DefaultBrandColor()
+
+        @available(*, deprecated, message: "Use DesignLibrary.Tokens")
+        public var appColors: AppColors = AppColors()
+
+        @available(*, deprecated, message: "Use DesignLibrary.componentBuilder")
+        public var modern: ModernPalette {
+            return ModernPalette(self)
+        }
+
+        fileprivate init(tokens: Tokens, traits: UITraitCollection) {
+            self.tokens = tokens
+            self.traits = traits
+        }
+
+        func font(fontAttributes: Tokens.Typography.FontAttributes) -> UIFont {
```

Comment:
Guidelines for factory methods recommend `makeFont(attributes:  Tokens.Typography.FontAttributes) -> UIFont`.